docs/development/doc_guide.rst

```diff
@@ -1,76 +1,633 @@
+
+Markdown
+--------
+
+A few of PlasmaPy's files are written using `Markdown
+<https://www.markdownguide.org/>`_, such as README files and licenses
+from other packages. Markdown is simpler but more limited than reST.
+Markdown files end with ``.md``. Posts on GitHub are written in
+`GitHub Flavored Markdown <https://github.github.com/gfm/>`_.
+The following code block contains a few common examples of Markdown
+formatting.
+
+.. code-block:: markdown
+
+   # Header 1
+
+   ## Header 2
+
+   Here is a link to [PlasmaPy's documentation](https://docs.plasmapy.org).
+
+   We can make text **bold** or *italic*.
+
+   We can write in-line code like `x = 1` or create a Python code block:
+
+   ```Python
+   y = 2
+   z = 3
+   ```
+
+Sphinx
+------
+
+`sphinx` is the software used to generate `PlasmaPy's online documentation`_
+from reST files and Python docstrings.
+
+Configuration
+~~~~~~~~~~~~~
+
+The `docs/conf.py`_ file contains the configuration information needed
+to customize `sphinx` behavior.
+The documentation for `sphinx` lists the `configuration options
+<https://www.sphinx-doc.org/en/master/usage/configuration.html>`_
+that can be set.
+
+Sphinx extensions
+~~~~~~~~~~~~~~~~~
+
+PlasmaPy documentation is built with the following `sphinx` extensions:
+
+* `sphinx.ext.autodoc
+  <https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html>`_
+  for including documentation from docstrings
+* `sphinx.ext.intersphinx
+  <https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html>`_
+  for linking to other projects' documentation
+* `sphinx.ext.graphviz
+  <https://www.sphinx-doc.org/en/master/usage/extensions/graphviz.html>`_
+  to allow `Graphviz <https://graphviz.org/>`_ graphs to be included
+* `sphinx.ext.mathjax
+  <https://www.sphinx-doc.org/en/master/usage/extensions/math.html#module-sphinx.ext.mathjax>`_
+  for math rendering with `MathJax <https://www.mathjax.org/>`_
+* `sphinx.ext.napoleon
+  <https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html>`_
+  for allowing NumPy style docstrings
+* `sphinx.ext.todo
+  <https://www.sphinx-doc.org/en/master/usage/extensions/todo.html>`_ to support
+  ``todo`` directives
+* `nbsphinx <https://nbsphinx.readthedocs.io>`_ for including
+  `Jupyter`_ notebooks
+* `sphinx_copybutton <https://sphinx-copybutton.readthedocs.io>`_ to add
+  a "copy" button for code blocks
+* `sphinx_gallery.load_style
+  <https://sphinx-gallery.github.io/stable/advanced.html?highlight=load_style#using-only-sphinx-gallery-styles>`_
+  for using sphinx-gallery styles
+* IPython.sphinxext.ipython_console_highlighting
+* `sphinx_changelog <https://sphinx-changelog.readthedocs.io>`_
+  for rendering `towncrier`_ changelogs
+* `plasmapy_sphinx` for customizations created for use in PlasmaPy.
+  Note that `plasmapy_sphinx` is expected to be broken out into its own
+  package at a future date.
+
+These extensions are specified in ``extensions`` in `docs/conf.py`_.
```

Comment:
```suggestion
These extensions are specified in :confval:`extensions` configuration
value in `docs/conf.py`_.
```

This should cross-link to the ``extensions`` documentation in sphinx.

Reply:
Actually, you should be able to do this with the `intersphinx_mappings` as well, ``:confvl:`intersphinx_mappings` ``.